packages/proxy/lib/http/response-middleware.ts

```diff
@@ -441,10 +475,34 @@ const MaybeRemoveSecurity: ResponseMiddleware = function () {
   this.next()
 }
 
-const GzipBody: ResponseMiddleware = function () {
-  if (this.isGunzipped) {
-    debug('regzipping response body')
-    this.incomingResStream = this.incomingResStream.pipe(zlib.createGzip(zlibOptions)).on('error', this.onError)
+const ReCompressBody: ResponseMiddleware = function () {
+  if (this.isUncompressed) {
+    const encodings: string[] = resGetContentEncodings(this.incomingRes)
+
+    debug('encodings to recompress: %o', encodings)
+    for (let encoding of encodings) {
+      let recompressor: zlib.Gzip | zlib.BrotliCompress | undefined
+
+      if (encoding === 'gzip') {
+        debug('recompressing gzip encoding')
```

Comment:
i know it's not the current pattern in this file, but can you change the debug statements you've added to `this.debug` instead of `debug`? `this.debug` will log some additional info about the current request